server/plugin/oAuth.go

```diff
@@ -108,11 +110,15 @@ func (p *Plugin) OAuthComplete(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	p.closeBrowserWindowWithHTTPResponse(w)
+	p.CloseBrowserWindowWithHTTPResponse(w)
 }
 
 // GenerateOAuthToken generates OAuth token after successful authorization
 func (p *Plugin) GenerateOAuthToken(code, state string) error {
+	if len(strings.Split(state, "_")) != 2 || strings.Split(state, "_")[1] == "" {
+		return errors.New("invalid state")
+	}
```

Comment:
why is this added here?
we are already checking this inside `OAuthComplete` on line `103`

Reply:
reverted